llvm/include/llvm/Support/MathExtras.h

```diff
@@ -385,6 +385,8 @@ inline uint64_t PowerOf2Ceil(uint64_t A) {
 ///   alignTo(~0LL, 8) = 0
 ///   alignTo(321, 255) = 510
 /// \endcode
+///
+/// May overflow.
```

Comment:
FYI there is still room for improvement here. Overflow is unavoidable if the result is not representable, but the current implementation also overflows in some cases where the correct result would be representable.

Examples with 8-bit integers:
alignTo(200, 128) = 0 which is OK because the true result 256 is not representable
alignTo(200, 127) = 0 but the true result 254 is representable

Reply:
Is it okay if I tackle this problem in the next patch (see #95426), since the numbers I need to test with in the current implementation are going to be very big and ugly?

Reply:
Sure. It is a pre-existing problem. No need to fix it right now.